back/lib/time_boundaries_parser.rb

```diff
@@ -7,8 +7,8 @@ def initialize(start_at, end_at)
   def parse
     timezone = AppConfiguration.instance.settings('core', 'timezone')
     platform_range = AppConfiguration.instance.created_at..Time.now.in_time_zone(timezone).end_of_day
-    start_range = @start_at&.to_date ? @start_at : platform_range.begin
-    end_range = @end_at&.to_date ? @end_at : platform_range.end
+    start_range = @start_at&.to_date ? @start_at.to_date : platform_range.begin
+    end_range = @end_at&.to_date ? @end_at.to_date : platform_range.end
```

Comment:
Yep, this code looks a bit weird. If the intention is to check for the presence of start/end dates, I would go for:
```suggestion
    start_range = @start_at ? @start_at.to_date : platform_range.begin
    end_range = @end_at ? @end_at.to_date : platform_range.end
```

or if we also expect empty strings (bad bad):
```suggestion
    start_range = @start_at.present? ? @start_at.to_date : platform_range.begin
    end_range = @end_at.present? ? @end_at.to_date : platform_range.end
```

Reply:
Yeah sounds good, will replace that :+1: